### PR TITLE
Fix(ci): Update deploy action

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR updates the used actions and combines the git steps.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -42,9 +42,10 @@ jobs:
           submodules: recursive
 
       - name: Set up JDK 1.8
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
-          java-version: 1.8
+          distribution: "temurin"
+          java-version: "17"
```
Comment (musicEnfanthen) [inline .github/workflows/deploy.yml:48]: Is ant running with it?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -42,9 +42,10 @@ jobs:
           submodules: recursive
 
       - name: Set up JDK 1.8
```
Comment (musicEnfanthen) [inline .github/workflows/deploy.yml]:
```suggestion
      - name: Set up Java

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -69,23 +70,15 @@ jobs:
       - name: Copy built schema to SCHEMA_DIR
         run: cp -r dist/schemata/dev $SCHEMA_DIR/
 
-      - name: Check git status before commit
```
Comment (musicEnfanthen) [inline .github/workflows/deploy.yml]: I know that these three steps are somewhat verbose, but it helps to keep things separated (also for debugging). So would prefer not to combine them.

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -111,25 +104,17 @@ jobs:
         run: |
           rm -rf $GUIDELINES_DIR/dev 
           cp -r dist/guidelines/dev/web $GUIDELINES_DIR/
-          mv $GUIDELINES_DIR/web $GUIDELINES_DIR/dev 
+          mv $GUIDELINES_DIR/web $GUIDELINES_DIR/dev
 
-      - name: Check git status before commit
```
Comment (musicEnfanthen) [inline .github/workflows/deploy.yml:116]: Same as above

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -24,13 +24,13 @@ env:
 
 jobs:
   build:
-
+    name: Build and deploy Schema
```
Comment (musicEnfanthen) [inline .github/workflows/deploy.yml]:
```suggestion
    name: Build and deploy Schema and Guidelines
```

**Review by musicEnfanthen — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (rettinghaus) [reply .github/workflows/deploy.yml:48]: Yes, tested.

Comment (musicEnfanthen) [reply .github/workflows/deploy.yml:48]: Great

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (musicEnfanthen) [reply .github/workflows/deploy.yml]: Thank you

**Review by musicEnfanthen — APPROVED** [review]

**Comment by bwbohl** [comment]
@rettinghaus what do you mean with 
> combines the git steps

can't really see a big change, although I think the small ones make a big difference ;-)

**Comment by bwbohl** [comment]
Moreover I'd rather see this merged to [pdf-guidelines-test ](https://github.com/music-encoding/music-encoding/tree/pdf-guidelines-test) which holds the latest developments of the build process

**Comment by musicEnfanthen** [comment]
@bwbohl This one updates the versions of different actions used right now (checkout, java setup, etc.). It should be fine to go for now in develop. The full script will be revised on pdf-guidelines-test for sure, but that is not primarily connected to this PR.

**Review by bwbohl — APPROVED** [review]

**Pull request merged**